airbrakes/data_handling/logger.py

```diff
@@ -123,55 +126,62 @@ def _convert_unknown_type(obj_type: Any) -> str:
 
-        index = 0  # Index to loop over processed data packets:
+        index = 0  # Index to loop over processor data packets:
         # Convert the imu data packets to a dictionary:
         for imu_data_packet in imu_data_packets:
             # Let's first add the state, extension field:
-            logged_fields: LoggedDataPacket = {
-                "state": state,
-                "extension": extension,
-            }
-            # Convert the imu data packet to a dictionary
+            logger_fields: LoggerDataPacket = {}
+            # Convert the packets to a dictionary
+            context_data_packet_dict: dict[str, int | str] = to_builtins(context_data_packet)
+            logger_fields.update(context_data_packet_dict)
+            servo_data_packet_dict: dict[str, int | float] = to_builtins(servo_data_packet)
+            logger_fields.update(servo_data_packet_dict)
             # Using to_builtins() is much faster than asdict() for some reason
             imu_data_packet_dict: dict[str, int | float | list[str]] = to_builtins(imu_data_packet)
-            logged_fields.update(imu_data_packet_dict)
+            logger_fields.update(imu_data_packet_dict)
 
-            # Get the processed data packet fields:
+            # Get the processor data packet fields:
             if isinstance(imu_data_packet, EstimatedDataPacket):
-                # Convert the processed data packet to a dictionary. Unknown types such as numpy
+                # Convert the processor data packet to a dictionary. Unknown types such as numpy
                 # float64 are converted to strings with 8 decimal places (that's enc_hook)
-                processed_data_packet_dict: dict[str, float] = to_builtins(
-                    processed_data_packets[index], enc_hook=Logger._convert_unknown_type
+                processor_data_packet_dict: dict[str, float] = to_builtins(
+                    processor_data_packets[index], enc_hook=Logger._convert_unknown_type
                 )
                 # Let's drop the "time_since_last_data_packet" field:
-                processed_data_packet_dict.pop("time_since_last_data_packet", None)
-                logged_fields.update(processed_data_packet_dict)
+                processor_data_packet_dict.pop("time_since_last_data_packet", None)
+                logger_fields.update(processor_data_packet_dict)
 
-                # Add the predicted apogee field:
-                logged_fields["predicted_apogee"] = predicted_apogee
                 index += 1
 
-            logged_data_packets.append(logged_fields)
+            # Apogee Prediction happens asynchronously, so we need to check if we have a packet
+            # This means that rather than belonging to a specific IMU packet, it belongs to the
+            # context packet
+            if apogee_predictor_data_packets:
+                apogee_predictor_data_packet_dict: dict[str, float] = to_builtins(
```

Comment:
Need a comment specifying that we only care about the last apogee predictor packet